compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -520,7 +520,14 @@ object desugar {
           enumCases.last.pushAttachment(DesugarEnums.DefinesEnumLookupMethods, ())
         val enumCompanionRef = TermRefTree()
         val enumImport =
-          Import(enumCompanionRef, enumCases.flatMap(caseIds).map(ImportSelector(_)))
+          Import(enumCompanionRef, enumCases.flatMap(caseIds).map{
+            enumCase => 
+              val zeroSpan = if enumCase.span.exists then
+                enumCase.span.endPos
+              else
+                enumCase.span
+              ImportSelector(enumCase.withSpan(zeroSpan)).withSpan(zeroSpan)
```

Comment:
Do we need both withSpan? Normally the outer span should be set based on the union of the inner spans.

Reply:
Actually, we just need to set it for `enumCase`. Good point!